src/boot/store.js

```diff
@@ -99,6 +99,7 @@ function dropCache(state: GlobalState): $Shape<GlobalState> {
     // $FlowFixMe[prop-missing]
     // $FlowFixMe[incompatible-variance]
     // $FlowFixMe[incompatible-type-arg]
+    // $FlowFixMe[incompatible-shape]
```

Comment:
Is this an appropriate thing to do? I have made the type `Debug = $Shape<{||}>` and flow was giving incompatible shape error on that, I am assuming here that we won't pass through `SessionState` (which contains `debug: Debug`) here because `storeKeys` don't contain that value and hence we can ignore the error related to that. Is that correct?

Reply:
Good question. I think the answer here is to not use `$Shape`:
* In general, `$Shape` has some funny unsound behavior (https://flow.org/en/docs/types/utilities/#toc-shape ) and so we usually avoid it.
* Here, `{||}` already means exactly the type of objects with no properties. So there isn't anything for `$Shape` to do, other than its funny unsoundness.

It looks like if I delete the `$Shape` I get an error at the point where it's initialized to `{}`. That is a known Flow quirk, where the empty object literal `{}` can't be used as the empty-object type `{||}`. Fortunately there is a clean workaround, which is to say `Object.freeze({})` instead: https://github.com/facebook/flow/issues/2386#issuecomment-695064325

Reply:
Good to know! thanks.